tests/test_multidict.py

```diff
@@ -387,6 +387,17 @@ def test_and2(self, cls: Type[MutableMultiMapping[str]]) -> None:
 
         assert {"key"} == {"key", "key2"} & d.keys()
 
+    def test_and_not_implemented(self, cls: Type[MutableMultiMapping[str]]) -> None:
+        d = cls([("key", "value1")])
+
+        with pytest.raises(TypeError, match=r"unsupported operand type(\(s\))? for \&"):
```

Comment:
It's usually better to match from start to finish:
```suggestion
        with pytest.raises(TypeError, match=r"^unsupported operand type(\(s\))? for \&$"):
```
(or at least, from start)

Reply:
But have you given any thought to implementing https://github.com/aio-libs/multidict/pull/936#discussion_r1473653123?

Reply:
`match=r"^unsupported operand type(\(s\))? for \|: '(multidict._multidict.)?_KeysView' and 'int'",`
is that better?
(`'(multidict._multidict.)?` included for c-ext tests)